### PR TITLE
add a flags in the report bandwidth

### PR DESCRIPTION
**Pull request by caberos**
https://github.com/softlayer/softlayer-python/issues/1419

### CODE REVIEW

File: SoftLayer/CLI/report/bandwidth.py [hunk 1]
```diff
@@ -213,24 +217,46 @@ def f_type(key, results):
...
+            item['type'],
+            item['name'],
+            formatting.b_to_gb(pub_in),
+            formatting.b_to_gb(pub_out),
+            formatting.b_to_gb(pri_in),
+            formatting.b_to_gb(pri_out),
+            item.get('pool') or formatting.blank(),
+        ])
+
+    if virtual:
```
Comment (allmightyspiff) [inline SoftLayer/CLI/report/bandwidth.py:236]: You should be using 1 Try/Except block here.

Something like this:


```python

try:
    if virtual:
        # get virtual bandwidth code
    elif hardware:
        # get hardware bandwidth code
    else:
        # do all 3 bandwidth things
except KeyboardInterrupt:
    env.err("Printing virtual collected results and then aborting.")



```

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Need to refactor the Try/Except blocks, otherwise looks good, thanks.

File: SoftLayer/CLI/report/bandwidth.py [hunk 2]
```diff
@@ -187,8 +187,12 @@ def _get_virtual_bandwidth(env, start, end):
 @click.option('--sortby', help='Column to sort by',
               default='hostname',
               show_default=True)
+@click.option('--virtual', is_flag=True, help='show the all bandwidth summary virtual',
```
Comment (ATGE) [inline SoftLayer/CLI/report/bandwidth.py:190]: maybe  `'show the all bandwidth summary for each virtual server'`

File: SoftLayer/CLI/report/bandwidth.py [hunk 3]
```diff
@@ -187,8 +187,12 @@ def _get_virtual_bandwidth(env, start, end):
...
+              default=False)
+@click.option('--server', is_flag=True, help='show the all bandwidth summary bare metal',
```
Comment (ATGE) [inline SoftLayer/CLI/report/bandwidth.py:192]: maybe  `'show the all bandwidth summary for each hardware server'`

**Review by ATGE — APPROVED** [review]
looks good

File: SoftLayer/CLI/custom_types.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
 import click
 
 
+# pylint: disable=inconsistent-return-statements
```
Comment (ATGE) [inline SoftLayer/CLI/custom_types.py:12]: try to disable  in the method convert as:
`def convert(self, value, param, ctx):  # pylint: disable=R1710`

File: setup.py [hunk 5]
```diff
@@ -4,6 +4,8 @@
 
 from setuptools import setup, find_packages
 
+# pylint: disable=inconsistent-return-statements
```
Comment (ATGE) [inline setup.py:7]: try to disable in the method `get_api_key` as:
`def get_api_key(client, username, secret):  # pylint: disable=R1710`

**Review by ATGE — APPROVED** [review]

**Review by allmightyspiff — APPROVED** [review]
Looks good to me, thanks

**Pull request merged**